src/main/java/seedu/address/model/recipe/Ingredient.java

```diff
@@ -43,20 +37,26 @@
      *
      * @param name A valid name.
      * @param quantity A valid quantity.
-     * @param quantifier A valid quantifier.
+     * @param quantifier A valid quantifier, blanks are allowed.
      */
     public Ingredient(String name, double quantity, String quantifier) {
         requireNonNull(name);
         checkArgument(isValidIngredientName(name), NAME_CONSTRAINTS);
 
         checkArgument(isValidQuantity(quantity), QUANTITY_CONSTRAINTS);
 
-        requireNonNull(quantifier);
-        checkArgument(isValidQuantifier(quantifier), QUANTIFIER_CONSTRAINTS);
+
+        if (quantifier == null) {
+            quantifier = "";
+        }
 
         this.ingredientName = name;
         this.quantity = quantity;
-        this.quantifier = quantifier;
+        this.quantifier = quantifier.trim();
```

Comment:
I think quantifier will already be trimmed? Might be an unnecessary call here but very minor tbh

Reply:
Similarly for the point above, if stub-testing with " ", or any other whitespace, it would instantiate with "".